packages/contracts-bedrock/src/cannon/PreimageOracle.sol

```diff
@@ -298,44 +296,26 @@ contract PreimageOracle is IPreimageOracle {
         // Pull storage variables onto the stack / into memory for operations.
         bytes32[KECCAK_TREE_DEPTH] memory branch = proposalBranches[msg.sender][_uuid];
         LPPMetaData metaData = proposalMetadata[msg.sender][_uuid];
+        uint256 blocksProcessed = metaData.blocksProcessed();
+
+        // The caller of `addLeavesLPP` must be an EOA.
+        if (msg.sender != tx.origin) revert NotEOA();
```

Comment:
Feels slightly risky that this moved from the very top of the function...

Reply:
No state changes can occur regardless, if we revert; Any specific concern?

Reply:
Oh yeah that's a good point. Ignore me then. :)